apis/config/v1beta1/configuration_types.go

```diff
@@ -352,3 +355,27 @@ type ClusterQueueVisibility struct {
 	// Defaults to 10.
 	MaxCount int32 `json:"maxCount,omitempty"`
 }
+
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enable indicates whether to enable fair sharing for all cohorts.
+	// Defaults to false.
+	Enable bool `json:"enable"`
+
+	// preemptionStrategies indicates which constraints should a preemption satisfy.
+	// The preemption algorithm will only use the next strategy in the list if the
+	// incoming workload (preemptor) doesn't fit after using the previous strategies.
+	// Possible values are:
+	// - LessThanOrEqualToFinalShare: Only preempt if the share of the preemptor CQ
+	//   will be less than or equal to the share of the preemptee CQ after the preemption occurs.
+	// - LessThanInitialShare: Only preempt if the share of the preemptor CQ will be strictly less
+	//   than the share of the preemptee CQ before the preemption.
```

Comment:
```suggestion
	// - LessThanOrEqualToFinalShare: Only preempt a workload if the share of the preemptor CQ
	//   is less than or equal to the share of the preemptee CQ without the workload.
	// - LessThanInitialShare: Only preempt a workload if the share of the preemptor CQ is strictly less
	//   than the share of the preemptee CQ with the workload.
```
Suggestion to say about preemption of a particular candidate workload rather than ` after the preemption occurs` (as in rule A). I think the current formulation may be confusing that we are talking about the end state of the preemption as a process (including all candidates), and it would not be accurate, because Rule B can add more workloads.

Reply:
I added more details and suggestions on when to use each strategy.